pkg/api/v1/atlascluster_types.go

```diff
@@ -17,23 +17,231 @@ limitations under the License.
 package v1
 
 import (
+	"github.com/jinzhu/copier"
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
+	"go.mongodb.org/atlas/mongodbatlas"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+func init() {
+	SchemeBuilder.Register(&AtlasCluster{}, &AtlasClusterList{})
+}
 
 // AtlasClusterSpec defines the desired state of AtlasCluster
 type AtlasClusterSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// ConnectionSecret is the name of the Kubernetes Secret which contains the information about the way to connect to
+	// Atlas (organization ID, API keys). The default Operator connection configuration will be used if not provided.
+	// +optional
+	ConnectionSecret *SecretRef `json:"connectionSecretRef,omitempty"`
+
+	// Collection of settings that configures auto-scaling information for the cluster.
+	// If you specify the autoScaling object, you must also specify the providerSettings.autoScaling object.
+	// +optional
+	AutoScaling *AutoScalingSpec `json:"autoScaling,omitempty"`
+
+	// Deprecated: do not use.
```

Comment:
FYI, if you try to use it your changes will be rejected, it's kept as users who had it enabled can still use it

Reply:
Forgot to delete this field - fixing